app/router.js

```diff
@@ -12,28 +12,56 @@ import {
   removeLabel,
   reopenIssue,
   requestReviewers,
+  reportError,
   transferIssue,
 } from "./github.js";
 import { getAuthToken } from "./auth.js";
 import { extractCommaSeparated, extractUsersAndTeams } from "./converters.js";
+import { labelEnabled, transferEnabled } from "./command-enabled.js";
+import { defaultConfig } from "./default-config.js";
+
+const { Octokit } = require("@octokit/core");
+const { config, composeConfigGet } = require("@probot/octokit-plugin-config");
 
 export async function router(auth, id, payload, verbose) {
   const sourceRepo = payload.repository.name;
   const transferMatches = transferMatcher(payload.comment.body);
   const actorRequest = `as requested by ${payload.sender.login}`;
+
+  const authToken = await getAuthToken(auth, payload.installation.id);
+  const octokit = new Octokit({ auth: authToken });
+
+  // TODO validate against schema
```

Comment:
I'd suggest not using what we used in release drafter.

There are better schema tools and I think it is better to keep the schema manually and not rely on tools that do not support the latest json schema.

Reply:
want to suggest one? basically I just want to not have to validate further down that someone hasn't stuck the wrong datatype in it etc or missing fields.

Reply:
https://github.com/ajv-validator/ajv or https://github.com/hyperjump-io/json-schema-validator as they both support the latest schema: https://json-schema.org/specification.html

There are tools that can generate from code but none that follow the latest schema.

Reply:
Ah right I always planned to write by hand, cheers for the links I’ve used ajv before so will probably try that first